url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/expert_system/Feel.java

```diff
@@ -0,0 +1,213 @@
+// ##################################################################
+// ##################################################################
+// # Feel Enum: 7 types to define rhythmic feels for responses		#						
+// ##################################################################
+// ##################################################################
+package expert_system;
+
+import java.util.ArrayList;
+
+
+public enum Feel {
+	
+	// ##########################################################
+	// #					 	  FIELDS						#
+	// ##########################################################
+	
+	STRAIGHTWALK,
+	SWUNGWALK,
+	STRAIGHTSOLO,
+	SWUNGSOLO,
+	LATIN,
+	LASTNOTE,
+	END;
+	
+	public ArrayList <Rhythmic_Block> fitRhythms;
+	
+	// ##########################################################
+	// #					   CONSTRUCTORS						#
+	// ##########################################################
+	private Feel(){
+		fitRhythms = new ArrayList<Rhythmic_Block>();
+		this.initialise();
+	}
+	// ##########################################################
+	// #					   	 METHODS						#
+	// ##########################################################
+	
+	// Returns fitRhythms field
+	public ArrayList<Rhythmic_Block> getFitRhythms() {
+		return fitRhythms;
+	}
+	
+	// Adds the appropriate rhythmic blocks to the fitRhythms arraylist
+	private void initialise(){
+		
+		String name = this.name();
+		
+		if(name == "END"){
+			fitRhythms.add(Rhythmic_Block.END);
+		}
+		else if(name == "LASTNOTE"){
+			fitRhythms.add(Rhythmic_Block.END);
+			fitRhythms.add(Rhythmic_Block.LASTNOTE);
+			
+		}
+		else if(name == "LATIN"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.